networkit/cpp/centrality/test/CentralityGTest.cpp

```diff
@@ -1953,4 +1954,125 @@ TEST_P(CentralityGTest, testGroupClosenessGrowShrink) {
+    g.addEdge(2, 3);
+    g.addEdge(2, 2);
+    g.addEdge(2, 4);
+    g.addEdge(3, 5);
+    g.addEdge(4, 5);
+    g.addEdge(5, 5);
+    g.addEdge(5, 6);
+    g.addEdge(5, 7);
+    g.addEdge(7, 7);
+
+    DegreeCentrality dc(g, false, true, false);
+    dc.run();
+
+    if (isDirected()){
+        EXPECT_EQ(2, dc.score(0));
+        EXPECT_EQ(1, dc.score(1));
+        EXPECT_EQ(3, dc.score(2));
+        EXPECT_EQ(1, dc.score(3));
+        EXPECT_EQ(1, dc.score(4));
+        EXPECT_EQ(3, dc.score(5));
+        EXPECT_EQ(0, dc.score(6));
+        EXPECT_EQ(1, dc.score(7));
+    } else {
+        EXPECT_EQ(2, dc.score(0));
+        EXPECT_EQ(1, dc.score(1));
+        EXPECT_EQ(5, dc.score(2));
+        EXPECT_EQ(2, dc.score(3));
+        EXPECT_EQ(2, dc.score(4));
+        EXPECT_EQ(6, dc.score(5));
+        EXPECT_EQ(1, dc.score(6));
+        EXPECT_EQ(2, dc.score(7));
+    }
+}
+
+TEST_P(CentralityGTest, testInDegreeCentrality) {
+    Graph g(8, false, isDirected());
+
+    g.addEdge(0, 2);
+    g.addEdge(0, 5);
+    g.addEdge(1, 2);
+    g.addEdge(2, 3);
+    g.addEdge(2, 2);
+    g.addEdge(2, 4);
+    g.addEdge(3, 5);
+    g.addEdge(4, 5);
+    g.addEdge(5, 5);
+    g.addEdge(5, 6);
+    g.addEdge(5, 7);
+    g.addEdge(7, 7);
+
+    DegreeCentrality dc(g, false, false, false);
+    dc.run();
+
+    if (isDirected()){
+        EXPECT_EQ(0, dc.score(0));
+        EXPECT_EQ(0, dc.score(1));
+        EXPECT_EQ(3, dc.score(2));
+        EXPECT_EQ(1, dc.score(3));
+        EXPECT_EQ(1, dc.score(4));
+        EXPECT_EQ(4, dc.score(5));
+        EXPECT_EQ(1, dc.score(6));
+        EXPECT_EQ(2, dc.score(7));
+    } else {
+        EXPECT_EQ(2, dc.score(0));
+        EXPECT_EQ(1, dc.score(1));
+        EXPECT_EQ(5, dc.score(2));
+        EXPECT_EQ(2, dc.score(3));
+        EXPECT_EQ(2, dc.score(4));
+        EXPECT_EQ(6, dc.score(5));
+        EXPECT_EQ(1, dc.score(6));
+        EXPECT_EQ(2, dc.score(7));
+    }
+}
+
+TEST_P(CentralityGTest, testDegreeCentralityIgnoreSelfLoops) {
+    Graph g(8, false, isDirected());
+
+    g.addEdge(0, 2);
+    g.addEdge(0, 5);
+    g.addEdge(1, 2);
+    g.addEdge(2, 3);
+    g.addEdge(2, 2);
+    g.addEdge(2, 4);
+    g.addEdge(3, 5);
+    g.addEdge(4, 5);
+    g.addEdge(5, 5);
+    g.addEdge(5, 6);
+    g.addEdge(5, 7);
+    g.addEdge(7, 7);
+
+    DegreeCentrality dc(g, false, true, true);
+    dc.run();
+
+    if (isDirected()){
+        EXPECT_EQ(2, dc.score(0));
+        EXPECT_EQ(1, dc.score(1));
+        EXPECT_EQ(2, dc.score(2));
+        EXPECT_EQ(1, dc.score(3));
+        EXPECT_EQ(1, dc.score(4));
+        EXPECT_EQ(2, dc.score(5));
+        EXPECT_EQ(0, dc.score(6));
+        EXPECT_EQ(0, dc.score(7));
+    } else {
+        EXPECT_EQ(2, dc.score(0));
+        EXPECT_EQ(1, dc.score(1));
+        EXPECT_EQ(4, dc.score(2));
+        EXPECT_EQ(2, dc.score(3));
+        EXPECT_EQ(2, dc.score(4));
+        EXPECT_EQ(5, dc.score(5));
+        EXPECT_EQ(1, dc.score(6));
+        EXPECT_EQ(1, dc.score(7));
+    }
+}
```

Comment:
Instead of writing a lot of `EXPECT_EQ` I recommend to create a `std::array` with the expected result and do the comparison in a loop (this applies to the previous tests as well).